erpnext/controllers/accounts_controller.py

```diff
@@ -922,9 +930,21 @@ def get_taxes_and_charges(master_doctype, master_name):
 				del tax[fieldname]
 
 		taxes_and_charges.append(tax)
-
 	return taxes_and_charges
 
+# remove existing Sales Taxes and Charges by its account head from taxes table
+def remove_existing_sales_taxes(doc_taxes):
+	sales_taxes_accounts = frappe.get_all("Sales Taxes and Charges", fields=["account_head"])
+	sales_taxes_names = [data['account_head'] for data in sales_taxes_accounts]
+
+	if doc_taxes:
+		to_remove = [d for d in doc_taxes if d['account_head'] in sales_taxes_names]
+
+		for d in to_remove:
+			doc_taxes.remove(d)
+
+	return doc_taxes
```

Comment:
commonify `remove_existing_sales_taxes` and `remove_existing_shipping_rule` functions.

Reply:
remove_existing_sales_taxes and remove_existing_shipping_rule functions are different.
the remove_existing_sales_taxes returns the taxes after removing the records & taxes get appended on the js side.
remove_existing_shipping_rule doesn't return any taxes & in this function, taxes get appended on the python side.